lighthouse-core/lib/i18n/i18n.js

```diff
@@ -473,6 +473,19 @@ function replaceIcuMessageInstanceIds(inputObject, locale) {
   return icuMessagePaths;
 }
 
+/** @typedef {import('./locales').LhlMessages} LhlMessages */
+
+/**
+ * Populate the i18n string lookup dict with locale data
+ * Used when the host environment selects the locale and serves lighthouse the intended locale file
+ * @see https://docs.google.com/document/d/1jnt3BqKB-4q3AE94UWFA0Gqspx8Sd_jivlB7gQMlmfk/edit
+ * @param {LH.Locale} locale
+ * @param {LhlMessages} lhlMessages
+ */
+function registerLocaleData(locale, lhlMessages) {
+  LOCALES[locale] = lhlMessages;
```

Comment:
I'm a little afraid to make `LOCALES` a mutable singleton like this now, but it's hard to think through the consequences and if they matter. #9296 and #9043 (and wherever the `field-performance` plugin feedback is) point at the fact that people likely will use this outside of devtools, though.

Reply:
> I'm a little afraid to make `LOCALES` a mutable singleton like this now, but it's hard to think through the consequences and if they matter. #9296 and #9043 (and wherever the `field-performance` plugin feedback is) point at the fact that people likely will use this outside of devtools, though.

if we don't want to expose this to anyone but devtools (I believe @exterkamp also preferred not to do that), another option (sorry I didn't think of until right now) is something like 
https://github.com/GoogleChrome/lighthouse/blob/b2dfb06221249d86dab667c241303b53c66203ed/clients/devtools-report-assets.js
basically rewriting `locales.js` to instead access the assets in the module. We could be clever instead of having to do it manually

Reply:
> We could be clever instead of having to do it manually

@connorjclark could even get to use a proxy to rewrite any getter on the `locales.js` object into a request for the resource :)